### PR TITLE
Fix empty spaces injected to Finance Loss

### LINKED ISSUE #2166: Extra spaces were wrongly inserted to string in "Other" at "How were your money or finances affected
## Summary
This is a bug of object handling and passing at "How were your money or finances affected"

> One paragraph explanation of the bug.
In the "How were your money or finances affected", there are many multiple selection, when user type in some thing like "AAAAAAA" in "Other".  when this information is passed to the end, it turns to "A A A A A A A"  

## Steps to reproduce
1) Follow GUI flow, when reaching "How were your money or finances affected", click any 2 of options including "Other", type in anything like "AAAAA"

2) Complete flow, in the end, "A A A A A" will be shown in the report.

> How exactly can the bug be reproduced? Be very specific.
Just follow above, it will be reproduced, found on master stream as of now 3:00PM July 31, 2020
This is very visible, user can easily found such information display disorder.

See attached pictures
HowWasFinanceAffected
SpaceInReport

## Unresolved questions

> Are there any related issues you consider out of scope for this issue that could be addressed in the future?
None

### PR DESCRIPTION
**Pull request by dengcn**
Fixes #2166 
This is to fix the issue of extra spaces were injected to the type in of "other" in financial loss section, and forcefully turned to lower case when user type in any (like "BBBBB") in other, the report showed "b b b b b "

# Description
This is to fix the issue of extra spaces were injected to the type in of "other" in financial loss section, and forcefully turned to lower case when user type in any (like "BBBBB") in other, the report showed "b b b b b "

> Please include a summary of the change.
const methods =
    data.moneyLost.methodOther && data.moneyLost.methodOther.length > 0
      ? data.moneyLost.methodPayment.concat([data.moneyLost.methodOther])
      : data.moneyLost.methodPayment
  //moneyLost.methodOther    
  // const paymentStringOrig = methods
  //   //.filter((method) => method !== 'methodPayment.other')
  //     .map((method) => unCamel(method.replace('methodPayment.', '')))
  //     .join(', ')
  //   // .filter((paymentStringOrig) => paymentStringOrig === 'other')
  
  let paymentStringOrig = methods
  let paymentString = String(paymentStringOrig).split('methodPayment.').join(' ').replace('other,', '')

# Any new packages installed?
No
> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [X ] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

**Comment by justinr86** [comment]
This issue assumes the user enters nonsense into the field. What happens if they type something like "my credit card was stolen"? Will that string be preserved?

**Comment by justinr86** [comment]
The function unCamel is used to format strings to a lower case version that will be easier for the analysts to read. For example, eTransfer becomes e transfer and creditCard becomes credit card. Your change removes this so the Financial Transactions portion is inconsistent with the rest of the report.

**Comment by justinr86** [comment]
Adding @ngosset to get his opinion.

**Comment by dengcn** [comment]
In case user type in ""my credit card was stolen", the code before will still inject space into this string. It would show "m y  c r e d i t  c a r d  w a s  s t o l e n" no matter user type in capital case or lower case

**Comment by justinr86** [comment]
> In case user type in ""my credit card was stolen", the code before will still inject space into this string. It would show "m y c r e d i t c a r d w a s s t o l e n" no matter user type in capital case or lower case

Are you sure about this? I tested multiple times on master branch and this did not happen.

**Comment by justinr86** [comment]
> In case user type in ""my credit card was stolen", the code before will still inject space into this string. It would show "m y c r e d i t c a r d w a s s t o l e n" no matter user type in capital case or lower case

**Comment by dengcn** [comment]
> > In case user type in ""my credit card was stolen", the code before will still inject space into this string. It would show "m y c r e d i t c a r d w a s s t o l e n" no matter user type in capital case or lower case
> 
> Are you sure about this? I tested multiple times on master branch and this did not happen.

The original picture in the PR description shows the issue, I am doing another mater branch run now using your string, will update.

**Comment by justinr86** [comment]
> > > In case user type in ""my credit card was stolen", the code before will still inject space into this string. It would show "m y c r e d i t c a r d w a s s t o l e n" no matter user type in capital case or lower case
> > 
> > 
> > Are you sure about this? I tested multiple times on master branch and this did not happen.
> 
> The original picture in the PR description shows the issue, I am doing another mater branch run now using your string, will update.

The image in the issue displays the result if the user enters garbage. This needs to be tested with a realistic string. The function unCamel will format the string to make it more readable. 

**creaditCard** -> **credit card**

The above transformation will be applied to the user entry using a regex. So if the user enters "I lost my credit card", it will appear as "i lost my credit card" in the report. However, if the user enters "AAAAA" it will appear as "a a a a a " because the same regex is applied. This is not ideal but is also not a bug, garbage in -> garbage out.

**Comment by dengcn** [comment]
I did more test, in this other field, 
-When user type in "my credit card was stolen", report shows "my credit card was stolen"  ==> OK
-When user type in "My Credit Card Was Stolen", report shows "my credit card was stolen"  ==> convert to lower case, not a big deal. OK
-When user type in "HHHHHHHHHHHH", report shows "h h h h h h h h h h h"  ==> This is where this PR is from, since there are other 3 similar places in the report.  all the strings in "others" in other 3 places are 100% reserved as is. 

In my opinion, for the user type in field string, we had better 100% reserve what user types in, leave them to the business analyst to decide/process user typed in information, under certain circumstances, pure user type-in may contain extra information which could bring the clue. It is more like what detective was going, this falls into business side. We technical side need to preserve all original information.

Attached please see report where it shows in other 3 places, this issue did not happen

====================================================================

What happened? | I WAS ASKED TO PAY $5000
-- | --
Amount of money asked for by suspect | 5555
Information asked for by suspect | credit card, dob, sin, CCCCCCCCCC, home address  ===>  "CCCCCC" 100% reserved
Amount of money stolen by suspect | 1111
Information obtained by suspect | credit card, dob, home address, sin, DDDDDDDDDDDD  ==> 100% reserved
Suspect name | CRA
-- | --
Suspect email | we@gmail.com
Suspect phone number | (613) 111-1111
Suspect website | www.cnn.com
Application used by the suspect | MS OFFICE
Suspect address | His account
Other means | AAAAAAAAA  ===> 100% reserved

Amount of money asked for by suspect | 5555
-- | --
Amount of money stolen by suspect | 1111
Method of Payment | e transfer, credit card, gift card, cryptocurrency, h h h h h h h h h h h h  ===> This is what this PR is about
Transaction date | 2020-03-05

**Comment by justinr86** [comment]
Your solution (removing unCamel) affects the other entries for Financial Information. In your example **e transfer, credit card, gift card, cryptocurrency** would become **eTransfer, creditCard, giftCard, cryptocurrency**. If the goal is to preserve the user's text, then only that field should be affected.

**Comment by dengcn** [comment]
> **e transfer, credit card, gift card, cryptocurrency** would become **eTransfer, creditCard, giftCard, cryptocurrency**

As for _"e transfer, credit card, gift card, cryptocurrency would become eTransfer, creditCard, giftCard, cryptocurrency", they will all be changed based on language(there is another PR 2082 which we were talking about cypress) once 2082 is merged in, all the string on report will be based on language toggle. And all the 4 others will behave the same.

**Comment by justinr86** [comment]
It is my opinion that the unintended change should be corrected before merging, even if another PR will fix it. This PR will introduce undesired functionality.

I will leave it up to @ngosset whether or not this should be merged.

**Comment by dengcn** [comment]
Update the code and do the best to keep old unCamel and old format for other types like e transfer, credit card, gift card, cryptocurrency.  

New code change has been pushed.
===============================================
  const methods = data.moneyLost.methodPayment

  const paymentStringOrig = methods
    .filter((method) => method !== 'methodPayment.other')
    .map((method) => unCamel(method.replace('methodPayment.', '')))
    .join(', ')

  let paymentString =
    String(paymentStringOrig) + ', ' + [data.moneyLost.methodOther]_
==================================================

The print on report is "e transfer, credit card, gift card, cryptocurrency, BBBBBBBBBB"
even though front of it will be changed in other PR

File: f2/src/ConfirmationSummary.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ export const ConfirmationSummary = () => {
     ...testdata.formData.anonymous,
     ...data.formData.anonymous,
   }
+  const { fyiForm } = data.formData
```
Comment (justinr86) [inline f2/src/ConfirmationSummary.js:39]: This PR appears to integrate changes that already exist in master such as FYI Flow. What happened?

Comment (dengcn) [reply f2/src/ConfirmationSummary.js:39]: I just did rebase yesterday, my change is just 1 file

Comment (dengcn) [reply f2/src/ConfirmationSummary.js:39]: Do not know why these pop up

**Comment by dengcn** [comment]
Since something was unexpectedly added to the difference during rebase, I created a new branch and new pull request, so we can close this one.

**Pull request closed without merging**